src/main/java/be/ugent/rml/Executor.java

```diff
@@ -375,14 +462,9 @@ private List<ProvenancedTerm> getIRIsWithTrueCondition(Record child, Term triple
             recordsMap.put("parent", parent);
 
             Object expectedBoolean = condition.execute(recordsMap);
-
-            if (expectedBoolean instanceof Boolean) {
-                if ((boolean) expectedBoolean) {
-                    ProvenancedTerm subject = this.getSubject(triplesMap, mapping, parent, i);
-                    iris.add(subject);
-                }
-            } else {
-                logger.warn("The used condition with the Parent Triples Map does not return a boolean.");
+            if (Boolean.TRUE.equals(expectedBoolean)) {
```

Comment:
This is faster than the original code?

Reply:
It is (or it can be depending on JRE).  The original did an instanceof check, a cast (= another instanceof check), a method call to a field access, and a boolean evaluation.  The JIT can inline the method call and likely reduce to just one instanceof check.

The optimized code is a method call to an object identity check plus a boolean evaluation.  This works because there will only ever be two instances of Boolean (TRUE + FALSE), allowing the object identity check to suffice (((Boolean) true) == ((Boolean) true)).  This effectively optimizes the instanceof check away, and only has impact on code called very often.

I don't recall the actual performance improvement.  It's less than anything else, especially after the use of the more efficient join significantly reduces how often this code is called.